File: LeetCode/Easy/1816-truncate-sentence/1816-truncate-sentence.go
func truncateSentence(s string, k int) string {
    n := len(s)
    i := 0

    for i < n {
        if s[i] == ' ' {
            k--
            if k == 0 {
                break
            }
        }
        i++
    }

    return s[:i]
}

File: LeetCode/Easy/1816-truncate-sentence/1816-truncate-sentence.java
class Solution {
    public String truncateSentence(String s, int k) {
        int n = s.length();
        int i = 0;
        
        while (i < n){
            char temp = s.charAt(i++);
            if (temp == ' ') {
                k--;
                if (k == 0) {
                    i--;
                    break;
                }
            }
        }
        
        return s.substring(0, i);
    }
}